### PR TITLE
usm: kafka: Change topic name size to u8

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

The topic name has an upper limit of 255 by the protocol. Also, we have a lower liimt of 80 characters. Thus, using u16 for the topic name size is a waste.

### Motivation

Reducing potential waste of memory.
Currently, does not impacting the map sizes as we have padding to align the struct.

### CODE REVIEW

File: pkg/network/protocols/kafka/statkeeper.go [hunk 1]
```diff
@@ -71,9 +71,9 @@ func (statKeeper *StatKeeper) GetAndResetAllStats() map[Key]*RequestStat {
 
 func (statKeeper *StatKeeper) extractTopicName(tx *KafkaTransaction) string {
 	// Limit tx.Topic_name_size to not exceed the actual length of tx.Topic_name
-	if tx.Topic_name_size > uint16(len(tx.Topic_name)) {
+	if uint16(tx.Topic_name_size) > uint16(len(tx.Topic_name)) {
```
Comment (DanielLavie) [inline pkg/network/protocols/kafka/statkeeper.go:74]: Shouldn't you cast `len(tx.Topic_name)` to `uint8` instaed?

**Review by DanielLavie — APPROVED** [review]
LGTM

File: pkg/network/protocols/kafka/statkeeper.go (hunk 1 above)
Comment (guyarb) [reply pkg/network/protocols/kafka/statkeeper.go:74]: to ensure we're not making a mistake by increasing topic_name ubffer beyond 255, I use uint16 cast hereto ensure we're not having "buffer overflow here"

**Pull request merged**